### PR TITLE
uniformint can now handle keyword arguments. 

### PR DESCRIPTION
**Pull request by daanklijn**
See issue https://github.com/hyperopt/hyperopt/issues/703.

### CODE REVIEW

**Comment by marctorsoc** [comment]
this LGTM, thanks for your contribution!

comments:
- could you add a test that fails without this fix? 
- it'd be great to explain a bit more why this fails in the description. `low` and `q` don't seem very related, so maybe it's more a problem of who defines args and kwargs
- please make sure the tests pass

**Comment by daanklijn** [comment]
Just added some tests. I placed them in `test_pyll_utils.py` as the `uniformint` is in  `pyll_utils.py`. Let me know if the test should be placed in a different file. 

For the explanation, I still think the problem relates to mistakenly switching around `low` and `q`.
Whenever someone calls `hp.uniformint` it will append the integer 1 to the `args` and pass the `args` and `kwargs`into `hp_quniform`.
```python
 def hp_uniformint(label, *args, **kwargs):
    args += (1.0,)
    return scope.int(hp_quniform(label, *args, **kwargs))
```
Eventually these `args` and `kwargs` end up in `stochastic.quniform` which expects `low`, `high` and `q`. Due to this structure the following things will happen for positional and keyword arguments:

`hp.uniformint("z", 0, 10)`: Whenever someone only uses `args`, `low` and `high` will still be the first and second arguments while `q` is the third argument, as it got appended later on. In this case the `uniformint` function is working perfectly fine.

`hp.uniformint("z", low=0, high=10)`: In this case the `args` array is initially empty and will contain only the value 1.0 after the appending of  (1.0,). As the `args` will be passed in first into `stochastic.quniform` it will assume that this 1 is the value of `low`. After that the `kwargs` are passed in and the function will throw a `TypeError` as it receives another value for the `low` argument.

File: hyperopt/tests/test_pyll_utils.py [hunk 1]
```diff
@@ -92,6 +94,21 @@ def stub_pyll_fn(label, low, high):
     pass
 
 
+class TestUniformintArguments(unittest.TestCase):
```
[the remark this answers is not in the document]
Comment (daanklijn) [reply hyperopt/tests/test_pyll_utils.py]: Thanks for the feedback! 👌  Just parameterized the tests. Tests pass locally but unfortunately Travis starts complaining about the number of arguments in my test function. See https://travis-ci.org/github/hyperopt/hyperopt/jobs/711780193#L1258. Any clue what might be causing this?

File: hyperopt/tests/test_pyll_utils.py [hunk 2]
```diff
@@ -92,6 +95,24 @@ def stub_pyll_fn(label, low, high):
...
+@pytest.mark.parametrize('arguments', [
+    ["z", 0, 10],
+    {"label": "z", "low": 0, "high": 10}
+])
+def test_uniformint_arguments(arguments):
+    """
+    Test whether uniformint can accept both positional and keyword arguments.
+    Related to PR #704.
+    """
+    if type(arguments) is list:
```
Comment (marctorsoc) [inline hyperopt/tests/test_pyll_utils.py:107]: suggestion: it's more standard to use `isinstance(arguments, list)`

**Comment by daanklijn** [comment]
Just made sure that `run_tests.sh` uses `pytest` now. Locally all tests besides the Spark tests pass (I don't have Spark setup currently). To be sure I tested the functionalities below and everything seems to work fine. Let me know what you think. 

`run_tests.sh`: runs all tests, including the Spark tests.
`run_tests.sh --no-spark`: runs all tests, excluding the Spark tests.
`run_tests.sh hyperopt/tests/test_webpage.py`: only runs the tests in `test_webpage.py`.
`run_tests.sh hyperopt/tests`: only runs the tests in the tests folder.

**Pull request merged**